src/sql/src/session/vars.rs

```diff
@@ -2733,10 +2733,10 @@ impl Drop for DropConnection {
 
 impl DropConnection {
     pub fn new_connection(
-        user: &User,
+        user: &str,
         active_connection_count: Arc<Mutex<ConnectionCounter>>,
     ) -> Result<Option<Self>, ConnectionError> {
-        Ok(if user.limit_max_connections() {
+        if !INTERNAL_USER_NAMES.contains(user) {
```

Comment:
@mjibson noted this was a behavior change [here](https://github.com/MaterializeInc/materialize/pull/25409/files#r1498097598).

Definitely worth discussing further. I don't understand why the current code exempts external admins from the connection limits. The max connections limit is to protect _us_, right? ISTM that we should not be allowing any end users to slip past the connection limit, even if they're an "Organization Admin".

Reply:
I believe the whole point of allowing external admins was so that they could come in and close the connections that were causing the limit to be reached (if say some long running connections got stuck for some reason in their infra). Just seems kinda bad to make the org admins sometimes not able to log in because non-admins are logged in? That was our reasoning at the time.

Reply:
I'm leaning towards retaining the original behavior, but definitely should add a comment because I didn't know the reasoning either!

Reply:
Ok, I figured out how to preserve the existing behavior. Sorry for the noise! I'll file an issue to follow up on the `max_connections` issue.